### PR TITLE
NRPT-654: implement ENV-EPD role controls in api and nrpti frontend

### PR DESCRIPTION
**Pull request by mtCarto**
Role controls for ENV-EPD role, includes necessary api and front-end updates.
- API: [NRPT-654](https://bcmines.atlassian.net/browse/NRPT-654)
- UI: [NRPT-353](https://bcmines.atlassian.net/browse/NRPT-353)

Assign yourself only the `admin:env-epd` role

POST to `/api/record` to test api record creation:

`{
    "orders": [
        {
            "recordName": "ENV EPD Role Test",
            "recordSubtype": "None",
            "issuingAgency": "Environmental Protection Division",
            "author": "BC Government",
            "issuedTo": {
                "type": "Individual",
                "companyName": null,
                "firstName": "Test",
                "middleName": "Middle",
                "lastName": "Name",
                "fullName": null,
                "dateOfBirth": null
            },
            "OrderNRCED": {
                "summary": "NRCED Summary",
                "addRole": "public"
            },
            "OrderLNG": {
                "description": "LNG Summary",
                "addRole": "public"
            }
        }
    ]
}`

Verify success and record read/write issuedTo read/write have **admin:env-epd** role

Create a document in the record:
`form-data
fileName: test-file-name.pdf
upfile: browse to your file`

Verify success document record has **admin:env-epd** role

To verify redaction, you can create a record in NRPTI from another user role and set the individual age to under 19. Then as the **admin:env-epd** role view the records list page and verify redaction (and vice versa for the roles).

### CODE REVIEW

File: api/src/controllers/document-controller.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const OBJ_STORE_URL = process.env.OBJECT_STORE_endpoint_url || 'nrs.objectstore.
 const ep = new AWS.Endpoint(OBJ_STORE_URL);
 const s3 = new AWS.S3({
   endpoint: ep,
+  sslEnabled: false,
```
Comment (marklise) [inline api/src/controllers/document-controller.js:17]: Keep ssl enabled.

File: api/src/controllers/post/permit.js [hunk 2]
```diff
@@ -145,6 +149,15 @@ exports.createMaster = function (args, res, next, incomingObj, flavourIds) {
...
 
+ // Add limited-admin(such as admin:wf) read/write roles if user is a limited-admin user
+ if (args) {
+  postUtils.setAdditionalRoleOnRecord(
+    permit,
+    args.swagger.params.auth_payload.realm_access.roles,
+    ADDITIONAL_ROLES
+  );
+}
+
```
Comment (marklise) [inline api/src/controllers/post/permit.js:160]: Indentation

File: api/src/controllers/document-controller.js (hunk 1 above)
Comment (mtCarto) [reply api/src/controllers/document-controller.js]: Removing, only for local testing.

**Review by marklise — APPROVED** [review]

**Review by BcGovNeal — CHANGES_REQUESTED** [review]
Permit flavour records are missing the ENV-EPD role and cause an error when editing Permit records

**Review by danieltruong — APPROVED** [review]

**Pull request merged**